services/skus/controllers.go

```diff
@@ -324,10 +325,6 @@ func handleSetOrderTrialDays(svc *Service) handlers.AppHandler {
 			return handlers.ValidationError("request", map[string]interface{}{"orderID": err.Error()})
 		}
 
-		if err := svc.validateOrderMerchantAndCaveats(ctx, orderID); err != nil {
-			return handlers.ValidationError("merchant and caveats", map[string]interface{}{"orderMerchantAndCaveats": err.Error()})
-		}
-
```

Comment:
This does not do anything useful for Premium orders. The endpoint is already authenticated. The merchant is always the same, it's `brave.com`, and we set it ourselves. And it seems like caveats for Premium are not populated at all, so the check is useless at least, and it also wastes two queries to the database to fetch the order.